client/index.js

```diff
@@ -22,41 +23,68 @@ socket.on('message', data => {
 
   case 'ADVENTURE_STATE':
     ReactDOM.render(<Adventure
-      uid={'/#' + socket.id}
-      character={data.characters.find(chr => chr.selectedBy === '/#' + socket.id)}
+      uid={uid}
+      character={data.characters.find(chr => chr.selectedBy === uid)}
       onActionSelect={id => createActionSubmit(emit, id)}
       />, mountNode);
     break;
 
   }
 });
 
-socket.on('connect', data => console.log('connect', data));
+socket.on('connect', handleConnect);
+socket.on('initialize', handleInitialize);
 socket.on('connect_error', err => console.error(err));
 socket.on('connect_timeout', err => console.error(err));
 
+function handleConnect() {
+  console.log('connect');
+  let uid = sessionStorage.getItem('__rock_in_my_boot_id__');
+  emit('initialize', {
+    uid,
+  });
+}
+
+function handleInitialize(data) {
+  console.log('initialized with uid', data);
+  sessionStorage.setItem('__rock_in_my_boot_id__', data.uid);
+  getCurrentState();
+}
+
 function getSocketUrl() {
   let [ protocol, url ] = document.location.origin.split(':');
   return `${protocol}:${url}:${process.env.WSS_PORT}`;
 }
 
+function getCurrentState() {
+  emit('message', {
+    type: 'CURRENT_STATE',
+  });
+}
+
 function createCharacterSelect(emit, id) {
+  let uid = sessionStorage.getItem('__rock_in_my_boot_id__');
   emit('message', {
     type: 'CHARACTER_CHOSEN',
     id,
+    uid,
   });
 }
 
 function createCharacterNegate(emit, id) {
+  let uid = sessionStorage.getItem('__rock_in_my_boot_id__');
   emit('message', {
     type: 'CHARACTER_NEGATE',
     id,
+    uid,
   });
 }
 
 function createActionSubmit(emit, id) {
+  let uid = sessionStorage.getItem('__rock_in_my_boot_id__');
```

Comment:
sessionStorage is limited to same-domain access, so this doesn't need to be so verbose (or we can use a constant)